### ISSUE TITLE
Error in canvas item shader: light is vec3 instead of vec4

### ISSUE THREAD

**Issue by BraindeadBZH** [godotengine/godot#13927]
in the file **/drivers/gles3/shaders/canvas.glsl** at line **288** in function **light_compute**:

The parameter **light** is defined as **vec3** which cause casting errors because it should be **vec4**.

**Comment by mrcdk** [comment]
Fixing this precise issue is easy, just changing a couple types in that line:

```glsl
void light_compute(inout vec4 light,vec2 light_vec,float light_height,vec4 light_color,vec2 light_uv,vec4 shadow,vec3 normal,vec2 uv,vec2 screen_uv,vec4 color) {
```

But I have no idea how the light shader works so I'm not sure if I should PR the small change or wait for someone else to fix it completly... if you use it on a `Light2D` node it doesn't seem to make any difference and if you use on another node then there are errors like:
```
ERROR: get_current_version: CanvasShaderGLES3: Fragment Program Compilation Failed:
0(284) : error C1008: undefined variable "shadow"
0(284) : error C1008: undefined variable "screen_uv"
```
Because, as far as I can tell, `shadow` is only available on `Light2D` nodes and for `screen_uv` to work you need to explicitly use `SCREEN_UV` on the shader (which is internally enabled on `Light2D` nodes)

**Comment by Jerome67000** [comment]
Up ! Since no one else has proposed something, maybe you can try to make a PR @mrcdk.

**Comment by reduz** [comment]
a PR is welcome

**Comment by akien-mga** [comment]
Fixed by #15636.

>  if you use it on a Light2D node it doesn't seem to make any difference and if you use on another node then there are errors like:
> ERROR: get_current_version: CanvasShaderGLES3: Fragment Program Compilation Failed:
0(284) : error C1008: undefined variable "shadow"
0(284) : error C1008: undefined variable "screen_uv"

That might be worth filing a new issue.
